pkg/applications/helmclient/client.go

```diff
@@ -0,0 +1,311 @@
+		return h.Install(chartLoc, releaseName, values)
+	}
+	return h.Upgrade(chartLoc, releaseName, values)
+}
+
+// Install the chart located at chartLoc into targetNamespace if the chart is already installed, an error is returned.
+// charLoc is the path to the chart archive (eg /tmp/foo/apache-1.0.0.tgz) or folder containning the chart (eg /tmp/mychart/apache).
+func (h HelmClient) Install(chartLoc string, releaseName string, values map[string]interface{}) (*release.Release, error) {
+	chartToInstall, err := h.buildDependencies(chartLoc)
+	if err != nil {
+		return nil, err
+	}
+
+	installClient := action.NewInstall(h.actionConfig)
+	installClient.Namespace = h.targetNamespace
+	installClient.ReleaseName = releaseName
+
+	rel, err := installClient.Run(chartToInstall, values)
+	if err != nil {
+		return nil, err
+	}
+	return rel, nil
+}
+
+// Upgrade the chart located at chartLoc into targetNamespace. if chart is not already installed an error is returned.
+// charLoc is the path to the chart archive (eg /tmp/foo/apache-1.0.0.tgz) or folder containning the chart (eg /tmp/mychart/apache).
+func (h HelmClient) Upgrade(chartLoc string, releaseName string, values map[string]interface{}) (*release.Release, error) {
+	chartToUpgrade, err := h.buildDependencies(chartLoc)
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeClient := action.NewUpgrade(h.actionConfig)
+	upgradeClient.Namespace = h.targetNamespace
+
+	rel, err := upgradeClient.Run(releaseName, chartToUpgrade, values)
+	if err != nil {
+		return nil, err
+	}
+	return rel, nil
+}
+
+// Uninstall the release in targetNamespace.
+func (h HelmClient) Uninstall(releaseName string) (*release.UninstallReleaseResponse, error) {
+	uninstallClient := action.NewUninstall(h.actionConfig)
+	return uninstallClient.Run(releaseName)
+}
+
+// buildDependencies add missing repositories and then do a Helm dependency build (ie download the chart dependencies
+// from repositories into "charts" folder).
+func (h HelmClient) buildDependencies(chartLoc string) (*chart.Chart, error) {
+	regClient, err := registry.NewClient() // todo credentials
+	if err != nil {
+		return nil, fmt.Errorf("can not initialize registry client %w", err)
+	}
+
+	var out strings.Builder
+	man := &downloader.Manager{
+		Out:              &out,
+		ChartPath:        chartLoc,
+		Getters:          h.getterProviders,
+		RepositoryConfig: h.settings.RepositoryConfig,
+		RepositoryCache:  h.settings.RepositoryCache,
+		RegistryClient:   regClient,
+		Debug:            false,
+		Verify:           downloader.VerifyNever,
+		SkipUpdate:       true,
+		// todo credentials
+	}
+
+	chartToInstall, err := loader.Load(chartLoc)
+	if err != nil {
+		return nil, fmt.Errorf("can not load char: %w", err)
+	}
+
+	fi, err := os.Stat(chartLoc)
+	if err != nil {
+		return nil, fmt.Errorf("can not find chart at `%s': %w", chartLoc, err)
+	}
+
+	// If we got the chart from the filesystem (ie cloned from a git repository). We have to do build dependencies because
```

Comment:
This should be one sentence. No full stop after the "(i.e. cloned from a git repository)"